### DISCUSSION TITLE
Append images, then cut to fit on pdf pages for printing.

### DISCUSSION THREAD

**Discussion by ballsystemlord** [ImageMagick/ImageMagick#4485] [Help] [answered]
Hello,
I've searched online and tried a few things without success.
What I want to do is to append several screenshots together vertically, then cut them up for printing outputting to a pdf file.
My command is:
`convert -append "../docs/Screenshot_2021-11-15"* -crop 1200x1550 +repage output.pdf`

I get output that's pdf and the images are concatenated like I want, but there's only one page.
I'm using imagemagick 6.9.XX.

Thanks

**Comment by snibgo** [comment]
First, put your command in the order you want it to be executed: 

Read the input files, then crop and repage, then append, then write the PDF.

or:

Read the input files, then append, then crop and repage, then write the PDF.

**Reply by ballsystemlord** [reply]
I chose B:
> Read the input files, then append, then crop and repage, then write the PDF.

`convert "../docs/Screenshot_2021-11-15"* -append -crop 1200x1550 +repage output.pdf`

Now I get 2 pages. I get all of the first image and part of the second on the second page.

I should have all of the images across at least 10 pages.

**Comment by snibgo** [comment]
How large is your image after the "-append"? To find out, you can insert "+write info:" (no quotes) at that point.

What are the specifications of the printer, ie how many pixels wide and high is each page? 

You can split your appended image into ten equal portions, if that will help, with "-crop x10@".

**Reply by ballsystemlord** [reply]
`1146x41856 1146x2221+0+0 8-bit sRGB 0.000u 0:00.009`
The printer prints on letter paper, which is 8.5in (21.59cm) wide by 11in (27.94cm) long. I sized how big the pages should be by using my pdf viewer to make a "best guess". I came up with 1200px by 1550px. Regarding how many pixels it can do, it's an inkjet, so it can do 1600 dots (pixels) per square inch -- but I really don't want to map the pixels to the dots as that would be too small.

Choosing equal portions (x30@ -- not x10@) is a "good enough" solution. But I'd really like to understand how to do it with the exact numbers I provided as it's much cleaner that way.

**Comment by snibgo** [comment, marked as answer]
You should insert "+repage" after "-append".

**Reply by ballsystemlord** [reply]
Oh neat!

Ok, so first you read the input files. Then you repage them. Then you crop them. Then you repage them again in their cropped form. Then you write to the output.

`convert "../docs/Screenshot_2021-11-15"* -append +repage -crop 1200x1550 +repage output.pdf`

**Comment by ballsystemlord** [comment]
So that I better understand imagemagick, why can't I append and repage them when they are read in? Why do all the options have to go after reading in the files?

**Comment by snibgo** [comment]
"-append" makes an output that is all the input images appended vertically. But it also sets the canvas to the size of the first input image. This is shown in your text output:
```
1146x41856 1146x2221+0+0 8-bit sRGB 0.000u 0:00.009
```
The image at that stage is 1146x41856 pixels, but on a canvas that is smaller, only 1146x2221. This messes with the crop. The cure is "+repage", which sets the canvas to the image size.

> why can't I append and repage them when they are read in?

You do. You read the input images, then append them to make a single image, then "+repage" to set the canvas of that image, then "-crop 1200x1550" to split into separate images, then "+repage" to set the canvas of each cropped image, then write all the images to output.pdf.

> Why do all the options have to go after reading in the files?

You should always write the options in the order you want them executed. IM V7 "magick" enforces that rule. IM v6 is more relaxed: you can specify the operations in any order, and IM v6 will guess what order you really want. For example, if you "-append" before you have any images that can be appended, v7 will say you can't do that, where v6 will guess that you really want to append _after_ you have read some images.

**Reply by ballsystemlord** [reply]
Being used to ffmpeg, when I saw this in the man page: `convert [input-option] input-file [output-option] output-file`
I assumed it would work like `ffmpeg -i 'concat:VID1.mp2|VID2.mp2' ...`.
So what I still need to understand is, why -append is not an input option?

**Comment by snibgo** [comment]
See the "-append" documentation at https://imagemagick.org/script/command-line-options.php#append :

"This option creates a single longer image, by joining all the current images in sequence top-to-bottom."

IM options are divided into _operations_ (which change images) and _settings_ (which affect what operations do).

A small number of options are either an operation or a setting, depending on context. For example "-colorspace" will change images into another colorspace or, if used before a PDF is read and rasterized, will affect what colorspace the PDF will be rasterized into.

Image operations can only be done once images have been read.
